store/sqlstore/user_store.go

```diff
@@ -188,6 +188,38 @@ func (us SqlUserStore) Update(user *model.User, trustedUpdateData bool) store.St
 	})
 }
 
+func (us SqlUserStore) AdminUpdateAuthData(userId string, authData *string, authService string, password string) store.StoreChannel {
```

Comment:
Is it necessary to add a new store function? I think using a bit of logic in the `app` function you can use either `UpdateAuthData` or `UpdatePassword` based on the inputs